src/ethereum/genesis.py

```diff
@@ -80,22 +79,28 @@ def get_genesis_configuration(genesis_file: str) -> GenesisConfiguration:
     ).decode()
     genesis_data = json.loads(genesis_str_data)
 
-    initial_balances = {
-        hex_to_bytes20(address): hex_to_u256(account["balance"])
-        for address, account in genesis_data["alloc"].items()
-    }
-
     return GenesisConfiguration(
         chain_id=U64(genesis_data["config"]["chainId"]),
         difficulty=hex_to_uint(genesis_data["difficulty"]),
         extra_data=hex_to_bytes(genesis_data["extraData"]),
         gas_limit=hex_to_uint(genesis_data["gasLimit"]),
         nonce=hex_to_bytes8(genesis_data["nonce"]),
-        timestamp=hex_to_u256(genesis_data["timestamp"]),
-        initial_balances=initial_balances,
+        timestamp=balance_str_to_u256(genesis_data["timestamp"]),
+        initial_accounts=genesis_data["alloc"],
     )
 
 
+def balance_str_to_u256(balance: str) -> U256:
```

Comment:
This function name is a misnomer since you're using it for the timestamp as well.